core/dns.go

```diff
@@ -122,13 +122,18 @@ func (dnsResolver *DNSResolverImpl) LookupHost(hostname string) ([]net.IP, time.
 	return addrs, aRtt, aaaaRtt, nil
 }
 
-// LookupCNAME sends a DNS query to find a CNAME record associated hostname and returns the
-// record target.
+// LookupCNAME returns the target name if a CNAME record exists for
+// the given domain name. If the CNAME does not exist (NXDOMAIN,
+// NXRRSET, or a successful response with no CNAME records), it
+// returns the empty string and a nil error.
 func (dnsResolver *DNSResolverImpl) LookupCNAME(hostname string) (string, time.Duration, error) {
 	r, rtt, err := dnsResolver.ExchangeOne(hostname, dns.TypeCNAME)
 	if err != nil {
 		return "", 0, err
 	}
+	if r.Rcode == dns.RcodeNXRrset || r.Rcode == dns.RcodeNameError {
+		return "", 0, nil
```

Comment:
I'll try to get to this with my StatsD cleanup.

On Tue, Jul 28, 2015 at 6:26 PM Tom Clegg <notifications@github.com> wrote:

> In core/dns.go
> <https://github.com/letsencrypt/boulder/pull/532#discussion_r35720670>:
>
> >  func (dnsResolver *DNSResolverImpl) LookupCNAME(hostname string) (string, time.Duration, error) {
> >  	r, rtt, err := dnsResolver.ExchangeOne(
>
> hostname, dns.TypeCNAME)
> >  	if err != nil {
> >  		return "", 0, err
> >  	}
> > +	if r.Rcode == dns.RcodeNXRrset || r.Rcode == dns.RcodeNameError {
> > +		return "", 0, nil
>
> Indeed. Fixed.
>
> It seems that miekg/dns can return meaningful rtt when err != nil too so
> we probably never have to zero it in Lookup*(). Stuck to just fixing
> these ones for now, though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/letsencrypt/boulder/pull/532/files#r35720670>.
>